python/ray/serve/api.py

```diff
@@ -466,52 +461,41 @@ def _run(
         raise TypeError(msg)
 
     parameter_group = []
-
     for deployment in deployments:
-        # Overwrite route prefix
-        if route_prefix is not DEFAULT.VALUE and deployment._route_prefix is not None:
-            if route_prefix is not None and not route_prefix.startswith("/"):
-                raise ValueError(
-                    "The route_prefix must start with a forward slash ('/')"
-                )
-
-            deployment._route_prefix = route_prefix
+        is_ingress = deployment._name == ingress_deployment_name
         if deployment.logging_config is None and logging_config:
-            if isinstance(logging_config, dict):
-                logging_config = LoggingConfig(**logging_config)
-            deployment.set_logging_config(logging_config.dict())
+            deployment = deployment.options(logging_config=logging_config)
+
         deployment_parameters = {
             "name": deployment._name,
             "replica_config": deployment._replica_config,
             "deployment_config": deployment._deployment_config,
             "version": deployment._version or get_random_string(),
-            "route_prefix": deployment.route_prefix,
-            "url": deployment.url,
+            "route_prefix": route_prefix if is_ingress else None,
             "docs_path": deployment._docs_path,
-            "ingress": deployment._name == ingress._name,
+            "ingress": is_ingress,
         }
         parameter_group.append(deployment_parameters)
+
     client.deploy_application(
         name,
         parameter_group,
         _blocking=_blocking,
     )
 
-    if ingress is not None:
-        # The deployment state is not guaranteed to be created after
-        # deploy_application returns; the application state manager will
-        # need another reconcile iteration to create it.
-        client._wait_for_deployment_created(ingress.name, name)
-        handle = client.get_handle(ingress.name, name, check_exists=False)
-        return handle
+    # The deployment state is not guaranteed to be created after
+    # deploy_application returns; the application state manager will
+    # need another reconcile iteration to create it.
+    client._wait_for_deployment_created(ingress_deployment_name, name)
+    return client.get_handle(ingress_deployment_name, name, check_exists=False)
 
 
 @PublicAPI(stability="stable")
 def run(
     target: Application,
     blocking: bool = False,
     name: str = SERVE_DEFAULT_APP_NAME,
-    route_prefix: Optional[str] = DEFAULT.VALUE,
+    route_prefix: Optional[str] = "/",
```

Comment:
Is there a reason why `route_prefix` had to default to `DEFAULT.VALUE` instead of `"/"` before?

Reply:
`None` means "no route prefix" and previously we wanted to pick up the route_prefix of the ingress deployment if nothing was specified here to override